namesys/routing.go

```diff
@@ -19,25 +23,74 @@ var log = logging.Logger("namesys")
 // routingResolver implements NSResolver for the main IPFS SFS-like naming
 type routingResolver struct {
 	routing routing.IpfsRouting
+
+	cache     *lru.Cache
+	cachelock sync.Mutex
+	cachelife time.Duration
 }
 
-// NewRoutingResolver constructs a name resolver using the IPFS Routing system
-// to implement SFS-like naming on top.
-func NewRoutingResolver(route routing.IpfsRouting) Resolver {
-	if route == nil {
-		panic("attempt to create resolver with nil routing system")
+func (r *routingResolver) cacheGet(name string) (path.Path, bool) {
+	r.cachelock.Lock()
+	ientry, ok := r.cache.Get(name)
+	r.cachelock.Unlock()
+
+	entry, ok := ientry.(cacheEntry)
+	if !ok {
+		return "", false
```

Comment:
The cache get `ok` is shadowed here.

Also: there is no differentiation here between a malformed and an elapsed entry. Could you insert a warning print?